framework/src/base/MooseApp.C

```diff
@@ -820,13 +820,19 @@ MooseApp::setupOptions()
       _builder.buildJsonSyntaxTree(tree);
     }
 
-    // Turn off live printing so that it doesn't mess with the dump
-    _perf_graph.disableLivePrint();
+    // Check if second arg is valid or not
+    if ((tree.getRoot()).is_object())
```

Comment:
when it is not valid, what is it? is it empty? Could we use that?

I m wondering if "is_object" might bite us later, when a valid search returned an array rather than an object for example.

Reply:
So it's not valid when the search parameter is not found in all the info --dump prints out. Having an empty param is valid because then it just prints out everything and calling is_object() on an empty parameter returns true. I used is_object() because that's what is called later on where the error ends up getting thrown. I believe it's called in json.h. I don't believe --dump can take more than one search parameter so we shouldn't have to worry about that right now, but if that does ever get changed then I will have to change how it checks for valid search parameters.

Reply:
An perhaps safer alternative to guard against non-objects could simply be:

```cpp
if (tree.getRoot().is_null())
  mooseError("Search parameter '", param_search, "' was not found in the registered syntax.");
```

Reply:
From my limited testing using .is_null() instead of .is_object() should work as well! The only potential issue I see is that later on .is_object() is still called. The ambiguous error was thrown from json.h from lines 21443 to 21454.

`    const_reference operator[](const typename object_t::key_type& key) const
    {
        // const operator[] only works for objects
        if (JSON_HEDLEY_LIKELY(is_object()))
        {
            auto it = m_data.m_value.object->find(key);
            JSON_ASSERT(it != m_data.m_value.object->end());
            return it->second;
        }


        JSON_THROW(type_error::create(305, detail::concat("cannot use operator[] with a string argument with ", type_name()), this));
    }
`
I just moved the is_object() call higher up so that I could return a clearer error. I didn't want to modify this section of code because I wasn't sure where else it might be used. But if there are any cases where is_null() and is_object() could both return false then we'd end up with the same error as before. If it's impossible for them to both return false then it shouldn't matter, especially since right now --dump can only handle one search string.